projects/spectator/src/lib/internals/key-parser.ts

```diff
@@ -0,0 +1,52 @@
+export interface ModifierKeys {
+  alt?: boolean;
+  control?: boolean;
+  shift?: boolean;
+  meta?: boolean;
+}
+
+export interface KeyOptions {
+  key: string | false;
+  keyCode: number | false;
+  modifiers: ModifierKeys;
+}
+
+export const parseKeyOptions = (keyOrKeyCode: string | number): KeyOptions => {
+  if (typeof keyOrKeyCode === 'number' && keyOrKeyCode) {
+    return { key: false, keyCode: keyOrKeyCode, modifiers: {} };
+  }
+
+  if (typeof keyOrKeyCode === 'string' && keyOrKeyCode) {
+    return parseKey(keyOrKeyCode as string);
+  }
+
+  throw new Error('keyboard.pressKey() requires a valid key or keyCode');
+};
+
+const parseKey = (keyStr: string): KeyOptions => {
+  if (keyStr.indexOf('.') < 0) {
+    return { key: keyStr, keyCode: false, modifiers: {} };
+  }
+
+  const keyParts = keyStr.split('.');
+  const key = keyParts.pop() as string;
+  const modifiers = keyParts.reduce((mods, part) => {
+    switch (part) {
+      case 'control':
+      case 'ctrl':
```

Comment:
In general, there is no need to return a new object on each iteration as you're starting with a new object.